yew/src/virtual_dom/vtext.rs

```diff
@@ -52,24 +52,27 @@ impl VDiff for VText {
         &mut self,
         _parent_scope: &AnyScope,
         parent: &Element,
-        next_sibling: Option<Node>,
+        mut next_sibling: Option<Node>,
         ancestor: Option<VNode>,
     ) -> Node {
-        if let Some(mut ancestor) = ancestor {
+        let _ancestor = if let Some(ancestor) = ancestor {
             if let VNode::VText(mut vtext) = ancestor {
                 self.reference = vtext.reference.take();
                 let text_node = self
                     .reference
                     .clone()
-                    .expect("text node should have reference");
+                    .expect("Rendered VText nodes should have a ref");
                 if self.text != vtext.text {
                     text_node.set_node_value(Some(&self.text));
                 }
                 return text_node.into();
             }
 
-            ancestor.detach(parent);
-        }
+            next_sibling = Some((&ancestor).into());
+            Some(super::DelayDetach { ancestor, parent })
```

Comment:
In some way, `next_sibling` is a pointer, and it becomes invalid as soon as `DelayedDetach` is dropped. We could define its lifetime as being valid until `_ancestor` is dropped. Our course, using it afterwards doesn't result in a segfault, but in correctness issue. So I wonder if there is a way to enforce it or advertise it a bit more. I am mainly worried about later refactoring that could go wrong.

Could we instead give the delayed dropping responsibility to next_sibling? Like creating an enum for next_sibling whose Drop implementation implements DelayedDetach for one of its variants? Or maybe that's too overkill?

Reply:
That's a good idea, I agree that this implementation is quite fragile. I'll play around with your ideas and see if I can get something working well

Reply:
I overhauled things so that we don't need to do this delayed dropping anymore